docs/setup-steps-1.4.0.md

```diff
@@ -0,0 +1,265 @@
+
+    Then `mosip-pms-client` should have any of the above roles.
+    
+4. Check that `mosip-pms-client` has the role `REGISTRATION_PROCESSOR`, `PARTNER_ADMIN`, `PMS_ADMIN` in Key Cloak. If this was not set by default, then set it and restart `keymanager` and `compliance-toolkit-service`.
+   
+5. It is also needed to generate an encryption key for CTK.
+   
+   *   Create a new app id by directly inserting the below row.
+
+       `INSERT INTO keymgr.key_policy_def(app_id, key_validity_duration, is_active,pre_expire_days, access_allowed, cr_by, cr_dtimes, upd_by, upd_dtimes, is_deleted, del_dtimes) VALUES ('COMPLIANCE_TOOLKIT', 1095, true, 60, 'NA', 'mosipadmin', '2022-11-28 09:00:40.822625', null, null, false, null);`
+       
+   *   Using the auth manager swagger URL, obtain the client token.
+
+       Swagger URL:
+
+       ```
+       `https://api-internal.${env}.mosip.net/v1/authmanager/swagger-ui/index.html?configUrl=/v1/authmanager/v3/api-docs/swagger-config`
+       ```
+
+       Endpoint:
+
+       ```
+       `/authenticate/clientidsecretkey`
+       ```
+
+       Request:
+
+       ```
+       {
+          "id": "string",
+          "version": "string",
+          "requesttime": "2022-12-22T07:13:35.010Z",
+          "metadata": {},
+          "request": {
+             "clientId": " mosip-pms-client ",
+             "secretKey": " XXXXXX ",
+             "appId": " partner "
+          }
+       }
+       ```
+   *   Now using the key manager swagger URL, generate module level certificate.
+
+       Swagger URL:
+
+       ```
+       `https://api-internal.${env}.mosip.net/v1/keymanager/swagger-ui/index.html?configUrl=/v1/keymanager/v3/api-docs/swagger-config`
+       ```
+
+       Endpoint:
+
+       ```
+       `/generateMasterKey/Certificate`
+       ```
+
+       Request:
+
+       ![](\_images/ctk-generateMasterkey.png)
+       
+   *   Directly download the certificate via key manager swagger URL and `getCertificate` endpoint, with App Id as `COMPLIANCE_TOOLKIT` and Ref Id as `COMP-FIR`.
+
+       ![](\_images/ctk-getCertificate.png)
+       
+   *   This certificate is to be used by **SBI** devices as the encryption key.
+
+       For Mock **MDS**, when running in **Auth** mode: update the below values in the `application.properties` file.
+
+       ```
+       mosip.auth.appid=regproc
+       mosip.auth.clientid=mosip-pms-client
+       mosip.auth.secretkey=XXXXXXXXXXXXXXXX
+       mosip.auth.server.url=https://api-internal.${env}.mosip.net/v1/authmanager/authenticate/clientidsecretkey 
+       mosip.ida.server.url=https://api-internal.${env}.mosip.net/v1/keymanager/getCertificate?applicationId=COMPLIANCE_TOOLKIT&referenceId=COMP-FIR
+       ```
+
+       For real MDS/SBI, the vendors can download the new encryption key from the UI and test with the updated **SBI** which uses this encryption key.It can be downloaded for **Auth SBI** projects from UI.
```

Comment:
For real MDS/SBI, the vendors can download the new encryption key from the UI and test with the updated **SBI** which uses this encryption key. It can be downloaded for **Auth SBI** projects from UI.